### PR TITLE
thread renaming on DFly

### PR DESCRIPTION
**Pull request by philmb3487**
Add support for thread renaming on DragonFlyBSD.
Sorry for the small commits, I just want to push what has been tested and it takes time.

### CODE REVIEW

File: src/util.cpp [hunk 1]
```diff
@@ -712,7 +712,7 @@ void RenameThread(const char* name)
 #if defined(PR_SET_NAME)
     // Only the first 15 characters are used (16 - NUL terminator)
     ::prctl(PR_SET_NAME, name, 0, 0, 0);
-#elif (defined(__FreeBSD__) || defined(__OpenBSD__))
+#elif (defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__DragonFly))
```
Comment (sipa) [inline src/util.cpp]: __DragonFly, not __DragonFly__ ?

Comment (laanwj) [inline src/util.cpp]: [This spec file]](http://ftp.netbsd.org/pub/pkgsrc/current/pkgsrc/lang/gcc46/files/dragonfly-spec.h) seems to imply it's `__DragonFly__`.

**Comment by sinetek** [comment]
It is. I am not sure what happened there, I must have pushed the wrong commit. Corrected.

**Comment by laanwj** [comment]
utACK after squashing the commits into one

**Pull request closed without merging**